lib/Search/PersonSearchProvider.php

```diff
@@ -0,0 +1,141 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\FaceRecognition\Search;
+
+use OCP\Search\IProvider;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\Search\ISearchQuery;
+use OCP\Search\SearchResult;
+use OCP\Search\SearchResultEntry;
+use OCP\Files\IRootFolder;
+
+use OCA\FaceRecognition\Db\Person;
+use OCA\FaceRecognition\Db\PersonMapper;
+use OCA\FaceRecognition\Db\ImageMapper;
+use OCA\FaceRecognition\Service\SettingsService;
+use OCA\FaceRecognition\Model\IModel;
+use OCA\FaceRecognition\Service\UrlService;
+
+/**
+ * Provide search results from the 'facerecognition' app
+ */
+class PersonSearchProvider implements IProvider {
+
+	/** @var PersonMapper personMapper */
+	private $personMapper;
+
+	/** @var ImageMapper imageMapper */
+	private $imageMapper;
+
+	/** @var SettingsService Settings service */
+	private $settingsService;
+
+	/** @var IL10N */
+	private $l10n;
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/** @var UrlService */
+	private $urlService;
+
+	/** @var IRootFolder */
+	private $rootFolder;
+
+	/** @var int*/
+	private $modelId;
+
+	public function __construct(PersonMapper      $personMapper,
+	                            ImageMapper       $imageMapper,
+	                            SettingsService   $settingsService,
+	                            IL10N             $l10n,
+	                            UrlService        $urlService,
+	                            IURLGenerator     $urlGenerator,
+	                            IRootFolder       $rootFolder) {
+		$this->personMapper     = $personMapper;
+		$this->imageMapper      = $imageMapper;
+		$this->settingsService  = $settingsService;
+		$this->l10n             = $l10n;
+		$this->urlService       = $urlService;
+		$this->urlGenerator     = $urlGenerator;
+		$this->rootFolder       = $rootFolder;
+		$this->modelId          = $this->settingsService->getCurrentFaceModel();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getId(): string {
+		return 'facerecognition';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getName(): string {
+		return $this->l10n->t('Face Recognition');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(string $route, array $routeParameters): int {
+		if ($route === 'settings.PersonalSettings.index' && $routeParameters["section"] === 'facerecognition') {
+			return 0;
+		}
+		return 10;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function search(IUser $user, ISearchQuery $query) : SearchResult {
+		$page = $query->getCursor() ?? 0;
+		$limit = $query->getLimit();
+		return SearchResult::paginated(
+			$this->l10n->t('Face Recognition'),
+			array_map(function (Person $result) {
+				$personName = $result->getName();
+				return new SearchResultEntry(
+				    $this->urlGenerator->imagePath('facerecognition','avatar.webp'),
+				    $personName,
+				    '',
+				    $this->urlService->getRedirectToPersonUrl($personName),
+				    '',
```

Comment:
```suggestion
				    'icon-facerecognition',
```
Acording:
https://github.com/nextcloud/server/blob/3ffd09d3f191def63ad2766368aaef62b333fcd1/lib/public/Search/SearchResultEntry.php#L86-L107

This parameter is the class of the icon... :wink: 

MM.. You must include the icon as:
https://github.com/matiasdelellis/facerecognition/blob/b26f39a1b47f7bfe52327548e61c87dba58c9ba7/css/facerecognition.scss#L5-L7

Reply:
```
 .icon-facrecognition { 
 	@include icon-color('app-dark', 'facerecognition', $color-black); 
 } 
```